hail/src/main/scala/is/hail/linalg/BlockMatrix.scala

```diff
@@ -1546,6 +1441,62 @@ private class BlockMatrixMultiplyRDD(l: BlockMatrix, r: BlockMatrix)
   @transient override val partitioner: Option[Partitioner] = Some(gp)
 }
 
+case class BlockMatrixRectanglesRDD(rectangles: Array[Array[Long]], bm: BlockMatrix)
+  extends RDD[(Int, BDM[Double])](bm.blocks.sparkContext, Nil) {
+
+  assert(rectangles.forall(rect => rect.length == 4))
+
+  val gp: GridPartitioner = bm.gp
+
+  override def compute(split: Partition, context: TaskContext): Iterator[(Int, BDM[Double])] = {
+    val rect = rectangles(split.index)
+    val rectStartRow = rect(0)
+    val rectEndRow = rect(1)
+    val rectStartCol = rect(2)
+    val rectEndCol = rect(3)
+
+    val rectData = new BDM[Double]((rectEndRow - rectStartRow).toInt, (rectEndCol - rectStartCol).toInt)
+    val blocksInRectangle = gp.rectangleBlocks(rect)
+    blocksInRectangle.foreach( blockIdx => {
```

Comment:
prefer `foreach { blockIdx =>` to `foreach(blockIdx => {`, they both pass a one-argument lambda, but the former has one less bracketing lexeme.